torch/distributed/device_mesh.py

```diff
@@ -538,132 +531,93 @@ def _setup_world_group_and_device(self):
 
-                    # If bound_device_id exists, it means the nccl communicator has been eagerly initialized
-                    # so that we can use `split_group` to create subgroups through `ncclCommSplit`.
-                    # In this case, we only need to make one API call (`split_group``) for the subgroup creation
-                    # for each mesh dimension. In a 2 * 4 mesh, we only need to make 2 API calls per ranks to create
-                    # all the subgroups.
-                    # Otherwise, we need to make more than one API call (`new_group`) for subgroup creations. The
-                    # numbers of API calls are equal to the number of subgroups for each mesh dimension. In a 2 * 4
-                    # mesh, we need to make 2 + 4 = 6 API calls per ranks to create all the subgroups.
-                    dim_group = None
-                    has_split_group = False
-                    if (
-                        (
-                            bound_device_id := getattr(
-                                default_group, "bound_device_id", None
-                            )
-                        )
-                        is not None
-                        and torch.cuda.is_available()
-                        and (
-                            backend is None
-                            or default_group._get_backend(torch.device("cuda")).name()
-                            == backend
-                        )
-                    ):
-                        dim_group = split_group(
-                            parent_pg=default_group,
-                            pg_options=pg_options,
-                            split_ranks=pg_ranks_by_dim.tolist(),
-                            group_desc=group_desc,
-                        )
-                        has_split_group = True
+            Returns:
+                DeviceMesh: A new DeviceMesh object with backend and layouts configured
 
-                    # If the subgroup has been already created through `split_group`, we simply loop over `pg_ranks_by_dim`
-                    # and append the `group_name` to the `dim_group_names` list when the current rank is in the subgroup.
-                    # Otherwise, we use `new_group` instead of `split_group` to create subgroups by looping over `pg_ranks_by_dim`
-                    # along with appending information to the `dim_group_names` list whenever necessary.
-                    for dim_mesh in pg_ranks_by_dim:
-                        subgroup_ranks = dim_mesh.tolist()
+            Note:
+                This is an internal method primarily used for creating submeshes when slicing
+                or transforming an existing DeviceMesh.
+            """
+            # Extract sizes and strides from layouts
+            size_l, stride_l, mesh_size = [], [], []
+            for layout in layouts:
+                size_l.extend(layout.sizes)
```

Comment:
i was expecting you to use layout algeba directly for this.  would 'compose' do the right thing here?  it doesn't seem like a good sign for 'layout' being the right tool for the job, if we have to convert the layout back to a tensor and iterate it that way for this operation..

Reply:
Switching to nested layout solves this problem.